modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
```

Comment:
This is probably rendering a `div` server-side because that's the base container.

Check other tags to see how to render a different base tag.

Reply:
You can check what you're rendering server side by disabling JS in a page and compare it with another one with JS enabled in the case the tag has hydration such as this one.

Reply:
I't s a div what we need to render here: https://github.com/liferay/clay/blob/aefb430bd83de2056122ae2af3fb06cae0b2f6ae/packages/clay-form/src/Checkbox.tsx#L64

Reply:
Do we want to replicate this DOM?
```
<div class="form-check">
	<input
		class="form-check-input"
		type="checkbox"
		value=""
		id="defaultCheck1"
	/>
	<label class="form-check-label" for="defaultCheck1">
		Default checkbox
	</label>
</div>
```

Or the one without the `div` wrapper, and do we keep the current state which is the following? 
```
<label>
	<input />
</label>
```

Reply:
We want to replicate this one https://github.com/liferay/clay/blob/aefb430bd83de2056122ae2af3fb06cae0b2f6ae/packages/clay-form/src/Checkbox.tsx#L64

Reply:
> I't s a div what we need to render here: https://github.com/liferay/clay/blob/aefb430bd83de2056122ae2af3fb06cae0b2f6ae/packages/clay-form/src/Checkbox.tsx#L64

Fair enough! Just make sure the attributes are all being rendered where we expect them to be ;)

Reply:
Good to know, thanks!